token_review.go

```diff
@@ -45,9 +44,7 @@ func tokenReviewAPIFactory(config *kubeConfig) tokenReviewer {
 	}
 }
 
-func (t *tokenReviewAPI) Review(ctx context.Context, jwt string, aud []string) (*tokenReviewResult, error) {
-
-	client := cleanhttp.DefaultClient()
+func (t *tokenReviewAPI) Review(ctx context.Context, jwt string, aud []string, client *http.Client) (*tokenReviewResult, error) {
 
 	// If we have a CA cert build the TLSConfig
 	if len(t.config.CACert) > 0 {
```

Comment:
I realized that we might get into concurrency issues here now that we're sharing the same `client` across requests (since this gets invoked on logins). Concurrent modification of the client in here might be problematic, so we'd have guard updates with a mutex write lock (and subsequently grab a read lock during the `Review` call). Now that we're storing the client in memory, we shouldn't have to set this on every review call; we can set the transport on instantiation of the client and on config updates/deletes.